src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -1082,35 +1064,25 @@ public void updateServiceTags(String serviceName, List<String> tags) {
 
     @Override
     public List<CloudServiceInstance> getServiceInstancesByMetadataLabelSelector(String labelSelector) {
-        Map<String, Metadata> serviceInstancesMetadata = getServiceInstancesMetadataByLabelSelector(labelSelector);
-        List<CloudServiceInstance> serviceInstances = fetchListWithAuxiliaryContent(() -> getServiceInstanceResourcesByNamesInBatches(serviceInstancesMetadata.keySet()),
-                                                                                    this::zipWithAuxiliaryServiceInstanceContent);
-        return getServiceInstancesWithMetadata(serviceInstances, serviceInstancesMetadata);
-    }
+        IntFunction<ListServiceInstancesRequest> listServiceInstancesPageRequestSupplier = page -> ListServiceInstancesRequest.builder()
```

Comment:
This variable name seems a bit too long. `pageSupplier` or `pageRequestSupplier` will be sufficient in my opinion